pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -91,6 +91,42 @@ type PingConfig struct {
 	Timeout time.Duration
 }
 
+// RPCConfig configures RPC serving.
+type RPCConfig struct {
+	// Network addresses to listen Neo RPC on. Each element must be a valid TCP
+	// address in 'host:port' format.
+	//
+	// Optional: by default, Neo RPC is not served.
+	Addresses []string
+
+	// Additional addresses that use TLS.
+	//
+	// Optional.
+	TLSConfig
+}
+
+// TLSConfig configures additional RPC serving over TLS.
+type TLSConfig struct {
+	// Additional TLS serving switcher.
+	//
+	// Optional: by default TLS is switched off.
+	Enabled bool
```

Comment:
even more stange to see it here, I'd determine extension by `Addresses` field emptiness

Reply:
we have `neo-go` [struct](https://github.com/nspcc-dev/neo-go/blob/d463bad96fa4dd6216f7609455d77d47ea4e4494/pkg/config/basic_service.go#L10) that should be filled. usually, i try to go 1:1 correspondence for better readability IMO (anyway we must fill the `Enabled` field on `neo-go` side)

Reply:
readability comes with duality when `Enabled == false` and `len(Addresses) == 0`? I guess no, lets just remove the flag and determine behavior by the state of one field - `Addresses`

Reply:
> that's where my own brain suffers

can you expand your suffering, please?

> readability comes with duality when Enabled == false and len(Addresses) == 0?

what duality is here? external lib (`neo-go`) requires `Enable` flag to be set, we have to use it and we agreed that we keep it in config files. if the prev state is correct, then we have `Enabled` in `yaml`, lose it in `Config` struct and dedicate it back for `neo-go` structs, what for?

> lets just remove the flag and determine behavior by the state of one field - Addresses

what if `enabled: false` and `listen: [*non-empty-list*]`? what `Addresses` field is? empty because of `enabled: false`? additional `if` checks? by the readability i meant: read the flag, put it in a middle struct if it is needed, take it to the `neo-go` lib. i do not like implicit transformations (one struct to another, to another, etc; one field to another (with another naming and another meaning), to another, etc) usually

Reply:
> lets resolve it first

sure

> overall, ur trying to get stuck in how Neo Go stuff is configured structurally

in fact, i am trying to make internal code as simple, as possible. i doubt our `blockchain` wrapper over the `neo-go` `blockchain` will be imported by anyone. i wanted to treat `Config` struct as an internal interpretation of `yml`/`json` config file: the one has `enabled` flag -- the other has too

> focus on docs and behavior

yes, i want the behavior to be the same as `neo-go` internals are. we import an incredibly complex thing from `neo-go` and it has the same config as we will provide (well, ok, that thread is not solved yet anyway). so why not just move it from `yml` to `blockchain` 1:1, the same way `neo-go` does in their own app?

Reply:
Gentlemen, the simpler the code for this stuff is the better. We do have `enabled` exposed externally in the node config. We do have it in NeoGo. Likely it'd be easier to avoid skipping/inferring it in the middle. I'd think that this way we get less transformations and less potential for errors. Or not?

Reply:
one field less anywhere and there is a simplification of the code in my understanding, I try not to produce dual conditions, but everyone has their own perception

let it remain as suggested while it works, we can always change